plugins/inputs/modbus/modbus_test.go

```diff
@@ -0,0 +1,261 @@
+package modbus
+
+import (
+	"testing"
+
+	m "github.com/goburrow/modbus"
+	"github.com/stretchr/testify/assert"
+	"github.com/tbrandon/mbserver"
+
+	"github.com/influxdata/telegraf/testutil"
+)
+
+type mbTest struct {
+	Name        string
+	Address     uint16
+	Quantity    uint16
+	WriteValue  []byte
+	ExpectValue interface{}
+}
+
+func TestCoils(t *testing.T) {
+	serv := mbserver.NewServer()
```

Comment:
It could be nice to drop the dependency, and creating a container running mbserver would be helpful for that.  If we go this route though then we should mock the client for the bulk of the tests and covering most code, and only use the container for broad integration tests.